### PR TITLE
feat: Scripture Updates

### PR DESCRIPTION
**Pull request by calebpanza**
## Motivation
In preparation for the launch of Daily Habits, we're giving the Scripture Feature a splash of paint. This is a first round of edits that will ultimately lead to us enabling Journals to be viewed/written directly on a Scripture.

### Scripture Feature
| Platform | Scripture Feature | Scripture Single |
| --- | --- | --- |
| iOS | Screen Shot 2022-07-25 at 12 31 27 PM | Screen Shot 2022-07-25 at 12 31 05 PM |
| iPadOS | Screen Shot 2022-07-25 at 1 48 48 PM | Screen Shot 2022-07-25 at 1 49 29 PM |
| Android | Screenshot_20220725-145621 | Screenshot_20220725-145936 |

### Scripture Journals
Below is a full screen grab on each platform for the Scripture Journals flow.

This was included inside of this PR since this PR was a dependency of this work. I know it grows the size of this PR quite a bit, but a lot of the files/changes are visuals and small interaction items.

**iOS**
https://user-images.githubusercontent.com/45076058/181784182-e4c0fdde-6473-41be-acae-6291d321e7d8.mp4

**Android**
https://user-images.githubusercontent.com/45076058/181784853-42a2398b-b0f4-4427-b5c1-bffa6bd9f904.mp4

### CODE REVIEW

**Comment by calebpanza** [comment]
Waiting for an API branch to get merged before marking this as ready.

I know there's a slight visual issue with the Android Scripture View, but it's minimal enough that I'm not gonna worry about it right now. During cooldown, I'll come back to it.

File: packages/apollos-ui-connected/src/hooks/useDeleteScriptureJournal.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      return;
+    }
+
+    const normalizedId = client.cache.identify({
+      id: scriptureId,
+      __typename: 'ScriptureJournal',
+    });
+
+    client.cache.evict({ id: normalizedId });
+    client.cache.gc();
```
Comment (vinnyjth) [inline packages/apollos-ui-connected/src/hooks/useDeleteScriptureJournal.js:43]: oh cool

File: packages/apollos-ui-connected/src/features/ScriptureFeature.js [hunk 2]
```diff
@@ -48,16 +102,33 @@ const ScriptureFeature = ({
       <PaddedView>{children}</PaddedView>
     );
   const navigation = useNavigation();
-  return scriptures.map((scripture) => (
+  return scriptures.map((scripture, i) => (
```
Comment (vinnyjth) [inline packages/apollos-ui-connected/src/features/ScriptureFeature.js:105]: what's `i` for?

File: packages/apollos-ui-connected/src/CommentListFeatureConnected/__snapshots__/CommentListFeatureConnected.tests.js.snap [hunk 3]
```diff
@@ -706,8 +706,8 @@ exports[`The CommentListFeatureConnected component should render 1`] = `
                   align="xMidYMid"
                   bbHeight={24}
                   bbWidth={24}
-                  color="rgba(39, 39, 46, 0.6)"
-                  fill="rgba(39, 39, 46, 0.6)"
+                  color="#c64f55"
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/CommentListFeatureConnected/__snapshots__/CommentListFeatureConnected.tests.js.snap:709]: Should there be changes to the comment list feature?

File: packages/apollos-ui-connected/src/ContentSingleFeaturesConnected/__snapshots__/ContentSingleFeaturesConnected.tests.js.snap [hunk 4]
```diff
@@ -735,3 +735,12 @@ exports[`ContentSingleFeaturesConnected should render with nodeId 1`] = `
   </View>
 </View>
 `;
+
+exports[`ContentSingleFeaturesConnected should render with nodeId 1`] = `
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/ContentSingleFeaturesConnected/__snapshots__/ContentSingleFeaturesConnected.tests.js.snap:739]: this snap doesn't look right

**Review by vinnyjth — CHANGES_REQUESTED** [review]
hey @calebpanza! looking really good overall. Two UX issues I ran into

- The "Finish" but

ton is hidden behind the journal screen. maybe we could could add some padding to the bottom of the scripture screen if there's a journal feature. 

- When I'm closing the journal feature, it's easy to accidentally close the entire scripture experience (I've actually never managed to close "just" the journal experience).

File: packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/AddJournal.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import isEmpty from 'lodash/isEmpty';
+
+import UserInput from './UserInput';
+import { ButtonLink } from '@apollosproject/ui-kit';
+
+const AddJournal = ({ navigation, route }) => {
+  const [userInput, setUserInput] = useState('');
+  const nextDisabled = isEmpty(userInput);
+
+  useLayoutEffect(() => {
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/AddJournal.js:21]: Admittedly not my code. Lol. Stole it from the docs on the bottom sheet module

File: packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  }, [profile, submitterIsCurrentUser]);
+
+  const contextMenu = useMemo(() => {
+    if (!submitterIsCurrentUser) {
+      return null;
+    }
+
+    return {
+      actions: [
+        { title: 'Edit Journal', systemIcon: 'square.and.pencil' },
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js:42]: Oh yes! Welcome to the magical world of system icons and SFSymbols! Lol. Seriously LOVE how Apple and Swift is handling iconography in their codebase.

File: packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    if (submitterIsCurrentUser) {
+      return 'Me';
+    }
+
+    if (profile?.currentUserFollowing?.state === 'ACCEPTED') {
+      return 'You follow';
+    }
+  }, [profile, submitterIsCurrentUser]);
+
+  const contextMenu = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js:35]: Love that idea!

**Comment by calebpanza** [comment]
@vinnyjth such great feedback! Thanks for catching those. I'm working through resolutions to them now.

File: packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            });
+            return;
+          case 1:
+            Alert.alert(
+              'Are you sure you want to delete your journal?',
+              'This action cannot be undone',
+              [
+                {
+                  text: 'Cancel',
+                  onPress: () => null,
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/ScriptureScreenConnected/ScriptureJournals/JournalFeed/CommentWithActions.js:63]: will this make the alert go away?

**Comment by calebpanza** [comment]
@vinnyjth here is a demo of those changes resolved. The code is a bit "cute" but it works! Lol

https://user-images.githubusercontent.com/45076058/183094344-83c77e3e-25ee-41b9-b27e-ec961d7d59e0.mp4

@redreceipt that method wasn't necessary and everything functions the same without it, so I removed it

**Comment by redreceipt** [comment]
snaps failing

**Comment by camrun91** [comment]
OK Couple of things I am seeing here. I am expecting touching the drawer to open I think I make this assumption because it looks and feels like a text box and I am anticipating input there. 
I would suggest making this part touchable and making it so it opens the drawer. What do you all think?
Screen Shot 2022-08-05 at 10 51 55 AM

**Comment by vinnyjth** [comment]
> I would suggest making this part touchable and making it so it opens the drawer. What do you all think?

I made the same mistake 😂  I agree it should be touchable.

**Comment by camrun91** [comment]
hahaha also make the text open up the editng area lol took me a bit to realize I had to hit the profile thing lol

**Comment by calebpanza** [comment]
https://user-images.githubusercontent.com/45076058/183116583-f29dbba3-c0db-4e45-859f-2deb8dea7b66.mp4

**Comment by camrun91** [comment]
@calebpanza what if when there is a journal that section says "see what others are saying." or something like that. Then touch that it opens the drawer to see the journals. Then when it is open tapping what stands out to you or profile icon pops you over to the entry view?

**Comment by calebpanza** [comment]
@camrun91 that's what it should be doing currently 😉 add a journal written by someone else in the dB to see all 3 states

**Comment by redreceipt** [comment]
text is hard to see in dark mode
Simulator Screen Shot - iPhone 13 - 2022-08-05 at 12 13 48

**Comment by camrun91** [comment]
> @camrun91 that's what it should be doing currently 😉 add a journal written by someone else in the dB to see all 3 states

right now the drawer only opens on swipe or touching the profile or the "input area" tapping either of those kicks you to the entry screen.

**Review by redreceipt — APPROVED** [review]
very nice :)

**Comment by redreceipt** [comment]
@vinnyjth have the changes you want been addressed?

**Comment by vinnyjth** [comment]
> @vinnyjth have the changes you want been addressed?

@redreceipt based on the videos I saw above, absolutely

**Review by vinnyjth — APPROVED** [review]

**Pull request merged**